packages/protocol/contracts/governance/Governance.sol

```diff
@@ -631,26 +633,125 @@ contract Governance is
       return false;
     }
 
+    require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
+
     address account = getAccounts().voteSignerToAccount(msg.sender);
-    Voter storage voter = voters[account];
     uint256 weight = getLockedGold().getAccountTotalLockedGold(account);
-    require(stage == Proposals.Stage.Referendum, "Incorrect proposal state");
     require(value != Proposals.VoteValue.None, "Vote value unset");
     require(weight > 0, "Voter weight zero");
-    VoteRecord storage voteRecord = voter.referendumVotes[index];
-    proposal.updateVote(
-      voteRecord.weight,
-      weight,
-      (voteRecord.proposalId == proposalId) ? voteRecord.value : Proposals.VoteValue.None,
-      value
+
+    Proposals.VoteValue[] memory currentValues = new Proposals.VoteValue[](1);
```

Comment:
I cannot do this since it is not possible to pass fixed-size array to function _vote which needs to accept arbitrary length array because it is being called also from votePartially function with different array length.

https://stackoverflow.com/questions/70128297/is-there-a-way-to-pass-fixed-array-as-parameter-to-utility-function-is-solidity

If I change this then I'm getting compilation error:

TypeError: Invalid type for argument in function call. Invalid implicit conversion from enum Proposals.VoteValue[1] memory to enum Proposals.

Reply:
updated according to our discussion